Frontend/ui-common/src/Components/Calendar/View.purs

```diff
@@ -290,28 +290,30 @@ dateView push config weekRow =
                             else if item.isStart && endDate.date /= 0 then Corners 50.0 true false false true
                             else if item.isStart && endDate.date == 0 then Corners 90.0 true true true true
                             else Corners 90.0 false false false false
-            , background $ if ((item.utcDate == startDate.utcDate) || (item.utcDate == endDate.utcDate)) && item.date /= 0 then Color.blue9000 
+            , background $ if (( compareTwoDates item startDate ) == "EQ" || (compareTwoDates item endDate ) == "EQ" ) && item.date /= 0 then Color.blue9000 
                              else if item.isInRange then Color.blue9000
                              else Color.transparent
             , onClick push $ const $ if (selectDateFlag item) then SelectDate (getSelectedDateRes item) else NoAction
             ][ textView
                [ width $ MATCH_PARENT
                , height $ MATCH_PARENT
                , text $ if item.date > 0 then show item.date else ""
-               , color $ if (item.utcDate < config.pastLimit.utcDate || item.utcDate > config.futureLimit.utcDate) then Color.black500
-                         else if ((item.utcDate == startDate.utcDate) || (item.utcDate == endDate.utcDate)) && item.date /= 0 then Color.white900
+               , color $ if item.date == 0 || (compareTwoDates item config.pastLimit == "LT" || compareTwoDates item config.futureLimit == "GT") then Color.black500
+                         else if item.date /= 0 && (( compareTwoDates item startDate ) == "EQ" || (compareTwoDates item endDate ) == "EQ" ) then Color.white900
                          else Color.black800
                , textSize FontSize.a_14
                , fontStyle $ FontStyle.regular LanguageStyle
-               , background $  if ((item.utcDate == startDate.utcDate) || (item.utcDate == endDate.utcDate)) && item.date /= 0 then Color.blue800
+               , background $  if (item.date /= 0) && (compareTwoDates item startDate == "EQ" || compareTwoDates item endDate == "EQ" ) then Color.blue800
                                else Color.transparent
                , gravity CENTER
-               , cornerRadii $ if (item.utcDate == startDate.utcDate) || (item.utcDate == endDate.utcDate) then Corners 90.0 true true true true
+               , cornerRadii $ if (item.date /= 0) && (( compareTwoDates item startDate ) == "EQ" || (compareTwoDates item endDate ) == "EQ" ) then Corners 90.0 true true true true
```

Comment:
This `(item.date /= 0) && (( compareTwoDates item startDate ) == "EQ" || (compareTwoDates item endDate ) == "EQ" )` can be put in let in

Reply:
done